hoprnet/hoprnet
# token authentication for backend

This is a part of #1890.

To test:

```
DEBUG="hopr*" yarn run:hoprd:alice --admin --apiToken 111
```

`curl -H -X POST -d "info" http://localhost:<rest_port>/api/v1/command` or 
`curl -H "X-Auth-Token: 222" -X POST -d "info" http://localhost:<rest_port>/api/v1/command` 

should give you 403.

And `curl -H "X-Auth-Token: 111" -X POST -d "info" http://localhost:<rest_port>/api/v1/command` should succeed.

To test websocket auth, you need `websocat`: https://github.com/vi/websocat

Try logging in with

`websocat ws://127.0.0.1:<admin_port>/ --header "Cookie:X-Auth-Token=bad_token`
`websocat ws://127.0.0.1:<admin_port>/ --header "Cookie:X-Auth-Token=111`
